src/Interpreters/InterpreterInsertQuery.cpp

```diff
@@ -682,7 +683,7 @@ QueryPipeline InterpreterInsertQuery::buildInsertSelectPipeline(ASTInsertQuery &
         presink_streams_size, sink_streams_size,
         table, /* view_level */ 0, metadata_snapshot, query_sample_block);
 
-    pipeline.resize(presink_chains.size());
+    pipeline.resize(presink_chains.size(), false, false, settings[Setting::enable_memory_based_pipeline_throttling]);
```

Comment:
lets try to use your memory resize in more cases. It would be good to try to use it always.